kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/internal/KafkaProxyFrontendHandler.java

```diff
@@ -171,6 +190,18 @@ public void userEventTriggered(
         else if (event instanceof AuthenticationEvent authenticationEvent) {
             this.authentication = authenticationEvent;
         }
+        else if (event instanceof SslHandshakeCompletionEvent sslHandshakeCompletionEvent) {
+            if (sslHandshakeCompletionEvent.isSuccess()) {
+                SslHandler sslHandler = getSslHandler(ctx);
+                try {
+                    downstreamCertificatePrincipal = sslHandler.engine().getSession().getPeerPrincipal().toString();
```

Comment:
Is `toString` the right choice here? [`Principal#getName`](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/security/Principal.html#getName()) looks like a more appropriate choice on my naive reading of the Javadoc.